commit 7d5ee6d9ac07c3823e476379cfc53e454a653e0a
Author: Razvan Cojocaru <rcojocaru@bitdefender.com>
Date:   Tue Dec 18 18:32:09 2018 +0200

    altp2m: always enable altp2m for all introspected guests

    Where introspection_extn is true, also enable altp2m support for
    all domains.

    Signed-off-by: Razvan Cojocaru <rcojocaru@bitdefender.com>

diff --git a/xen/arch/x86/hvm/hvm.c b/xen/arch/x86/hvm/hvm.c
index cf2caaa..6c3ab0f 100644
--- a/xen/arch/x86/hvm/hvm.c
+++ b/xen/arch/x86/hvm/hvm.c
@@ -626,6 +626,10 @@ int hvm_domain_initialise(struct domain *d)
     BUILD_BUG_ON(NR_HVM_DOMU_IRQS < NR_ISAIRQS);
     ASSERT(hvm_domain_irq(d)->nr_gsis >= NR_ISAIRQS);
 
+    /* Always enable altp2m support for introspected guests. */
+    if ( opt_introspection_extn )
+        d->arch.hvm.params[HVM_PARAM_ALTP2M] = XEN_ALTP2M_external;
+
     /* need link to containing domain */
     d->arch.hvm.pl_time->domain = d;
 
